### PR TITLE
Add PR + Issues templates

### LINKED ISSUE #627: create a contributor issue and PR template
To facilitate feedback requests and aid in getting new contributions in faster, we can provide issue and pull request templates. These should get information from someone who wants to contribute something, found something broken or missing, etc.

The PR template could highlight some things from our style and contributing guide to be sure they were addressed before opening a PR.

### PR DESCRIPTION
**Pull request by thomasyu888**
Not sure what else to add to this, but please feel free to add to this.
fixes #627

### CODE REVIEW

**Comment by Aryllen** [comment]
This is a personal preference so I would like @kelshmo to weigh in, as well. I tend to prefer templates that are not meant to be deleted before submission, but rather be more of a questionnaire/checklist style that should be filled out and kept in the PR. I think this helps in a couple ways. 

1. The person reviewing the PR can see if the person ran it locally, looked at the contributing guide, etc. 
2. It makes it harder to ignore and makes the person a bit more accountable.

My suggestion would be to remove the request to delete the template and leave the checklist (maybe add verification that they've looked at the guide). The question in the checklist about there being an issue could be changed to something like "did you reference the GH issue being addressed by number?" 

Again, these are my preferences and I would defer to @kelshmo's ideas since she is doing the most work on this project.

**Comment by thomasyu888** [comment]
I think I agree with that. An example of a nice issues from templates can be found in pandas GitHub: https://github.com/pandas-dev/pandas/issues

**Comment by Aryllen** [comment]
> An example of a nice issues from templates can be found in pandas GitHub: https://github.com/pandas-dev/pandas/issues

Oh, that is nice!

**Comment by thomasyu888** [comment]
Thanks @Aryllen ,  I made some edits.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- [ ] Did you reference the issue being addressed by number? (fixes #xxxx)
```
Comment (thomasyu888) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
- [ ] fixes #xxxx (Link Github Issue if appropriate?)
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- [ ] fixes #xxxx (Link Github Issue if appropriate?)
+- [ ] Did you examine your build locally (instructions [here](https://help.github.com/en/github/working-with-github-pages/testing-your-github-pages-site-locally-with-jekyll)?
```
Comment (thomasyu888) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion
- [ ] Did you examine your build locally (instructions [here](https://help.github.com/en/github/working-with-github-pages/testing-your-github-pages-site-locally-with-jekyll))?
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+- [ ] fixes #xxxx (Link Github Issue if appropriate?)
+- [ ] Did you examine your build locally (instructions [here](https://help.github.com/en/github/working-with-github-pages/testing-your-github-pages-site-locally-with-jekyll))?
```
Comment (thomasyu888) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+- [ ] fixes #xxxx (Link Github Issue if appropriate?)
+- [ ] Did you add a good description about your changes or additions?
```
Comment (thomasyu888) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion
- [ ] If you are creating a page, did you follow the instructions [here](https://github.com/Sage-Bionetworks/synapseDocs#creating-a-page)?
- [ ] Did you follow the [style guide](https://github.com/Sage-Bionetworks/synapseDocs#style-guide)?
- [ ] Did you add a good description about your changes or additions?
```

**Comment by Aryllen** [comment]
I noticed that you removed the bit about running locally. You could potentially keep that, but make a note that it isn't required (we don't want to discourage people from contributing). Personally, I think it helps to know if someone actually tried it locally (I expect less build errors).

**Review by kelshmo — APPROVED** [review]

**Pull request merged**